content/guides/advanced/testing-with-jest-and-bankrun.md

```diff
@@ -0,0 +1,372 @@
+---
+featured: false
+date: 2024-08-08T00:00:00Z
+difficulty: intermediate
+title: "Speed up Solana program tests with Jest and Bankrun"
+description:
+  "Testing programs is important. Bankrun offers a lightweight alternative to
+  the local validator for testing Solana programs and enables features like
+  custom account data and time travel. Tests can be up to 10 times faster using
+  bankrun."
+tags:
+  - typescript
+  - testing
+keywords:
+  - tutorial
+  - testing
+  - bankrun
+  - jest
+  - intro to solana development
+  - blockchain developer
+  - blockchain tutorial
+  - web3 developer
+---
+
+# Testing Solana Programs with Jest and Bankrun
+
+Testing your Solana programs is a critical part of the development process to
+ensure that your program behaves as expected and can even speedup your
+development. This guide will walk you through how you can test your Solana
+programs.
+
+Most Solana tests use the [Mocha framework](https://mochajs.org/) for writing
+the tests and [Chai](https://www.chaijs.com/) for assertions. However, you can
+use any testing framework that you are comfortable with. In this guide we will
+have a look at [Jest](https://jestjs.io/) and
+[Bankrun](https://kevinheavey.github.io/solana-bankrun/). With Bankrun, you can
+accelerate your tests by almost 10x, gain the ability to modify program time,
+and write custom account data.
+
+## Presets
+
+There are a few presets that will set you up with a basic testing environment
```

Comment:
Can you add in an quick explanation of the differences in the presets?

Reply:
Added :)